packages/svelte/src/compiler/migrate/index.js

```diff
@@ -586,6 +602,93 @@ const template = {
 		handle_events(node, state);
 		next();
 	},
+	SvelteComponent(node, { state, next, path }) {
+		let expression = state.str.original.substring(
+			/** @type {number} */ (node.expression.start),
+			node.expression.end
+		);
+
+		if (state.analysis.uses_props || state.analysis.uses_rest_props) {
+			let difference = 0;
+			// we need to migrate $$props and $$restProps by hand to update all the $$props and $$restProps in the expression
+			walk(
+				node.expression,
+				{},
+				{
+					Identifier(id_node, { path, next }) {
+						const parent = path[path.length - 1];
+						if (id_node.name !== '$$props' && id_node.name !== '$$restProps') return;
+
+						if (parent.type === 'MemberExpression' && parent.object !== id_node) return;
+
+						const props_name =
+							id_node.name === '$$props' || state.analysis.uses_props
+								? state.names.props
+								: state.names.rest;
+						const props_start =
+							/** @type {number} */ (id_node.start) - /** @type {number} */ (node.expression.start);
+
+						expression =
+							expression.substring(0, props_start + difference) +
+							props_name +
+							expression.substring(props_start + id_node.name.length + difference);
+
+						difference += props_name.length - id_node.name.length;
+						next();
+					}
+				}
+			);
```

Comment:
Is it possible to simplify this by calling `next` on `SvelteComponent` first, thereby transforming `$$props` etc through the other visitors, and then do a `str.move(..)` of the `this` binding to the correct position (to the `$derived` or `@const`)?

Reply:
Uhhhh here's shows my partial lack of knowledge of `magic-string`...i'll check and eventually simplify!

Reply:
@dummdidumm i was not able to use move but actually i found something i was initially searching for which is `snip` so i can get the updated expression directly! It's even better